tests/runner/src/main/java/org/mule/test/runner/classloader/TestContainerClassLoaderFactory.java

```diff
@@ -113,7 +121,11 @@ protected ArtifactClassLoader createArtifactClassLoader(final ClassLoader parent
    * @return uses only the set of {@link URL}s defined for the container to create the {@link ClassLoaderLookupPolicy}
    */
   public ClassLoaderLookupPolicy getContainerClassLoaderLookupPolicy() {
-    return withContextClassLoader(classLoader, () -> super.getContainerClassLoaderLookupPolicy(discoverModules()));
+    try (URLClassLoader classLoader = new URLClassLoader(urls, null)) {
```

Comment:
Doesn't this change have an impact on performance? before this was being created once for this object and now it is on each method call.